### ISSUE TITLE
Showing pandas dataframe crashes

### ISSUE THREAD

**Issue by vtraag** [spyder-ide/spyder#17455]
### What steps will reproduce the problem?

```python
import pandas as pd
df = pd.DataFrame({'index': [0, 1],
                   'var': ['test 1', 'test 2']})
```

Then show the `df` from the Variable Explorer.

### Traceback
```python-traceback
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
TypeError: invalid result from DataFrameHeaderModel.headerData(), unable to convert a Python 'Alignment' object to a C++ 'Qt::Alignment' instance
```

## Versions

* Spyder version: 5.2.2 
* Python version: 3.10.2
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Linux 5.11.0-46-generic

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0;<8.0.0        :  7.32.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.4.0 (OK)
keyring >=17.0.0              :  21.2.1 (OK)
nbconvert >=4.0               :  6.4.2 (OK)
numpydoc >=0.6.0              :  1.2 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0                :  2.12.2 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.3.2;<1.4.0          :  1.3.3 (OK)
pylsp_black >=1.0.0           :  1.1.0 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.1 (OK)
qtconsole >=5.2.1;<5.3.0      :  5.2.2 (OK)
qtpy >=1.5.0                  :  2.0.1 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  60.9.3 (OK)
sphinx >=0.6.6                :  4.3.2 (OK)
spyder_kernels >=2.2.1;<2.3.0 :  2.2.1 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.27 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=2.0.0            :  3.5.1 (OK)
numpy >=1.7                   :  1.22.3 (OK)
pandas >=1.1.1                :  1.4.1 (OK)
scipy >=0.17.0                :  1.8.0 (OK)
sympy >=0.7.3                 :  None (NOK)
```

**Comment by ccordoba12** [comment]
Hey @vtraag, thanks for reporting. This is a duplicate of issue #17361 and it'll be fixed in our next version (5.3.0), to be released later this month.

In the meantime, please use Spyder with Python 3.9.

**Comment by ccordoba12** [comment]
No problem, that's fine.
